src/main/java/org/breedinginsight/model/delta/DeltaEntity.java

```diff
@@ -0,0 +1,43 @@
+package org.breedinginsight.model.delta;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+
+import java.util.UUID;
+
+public abstract class DeltaEntity<T> {
+
+    protected final Gson gson;
+
+    @NonNull
+    protected final T brAPIObject;
```

Comment:
The name should be changed from `brapiObject` to `entity` since some delta entities will wrap around a system object. DeltaLocation for example will store a reference to an instance of type `ProgramLocatiion`.